src/components/funding-account/funding-account.service.ts

```diff
@@ -0,0 +1,258 @@
+import { Injectable, NotFoundException } from '@nestjs/common';
+import { node, relation } from 'cypher-query-builder';
+import { DateTime } from 'luxon';
+import { DuplicateException, ISession, ServerException } from '../../common';
+import {
+  addAllSecureProperties,
+  addPropertyCoalesceWithClause,
+  addUserToSG,
+  ConfigService,
+  createBaseNode,
+  DatabaseService,
+  filterByString,
+  ILogger,
+  Logger,
+  matchRequestingUser,
+  matchUserPermissions,
+  OnIndex,
+  runListQuery,
+} from '../../core';
+import {
+  CreateFundingAccount,
+  FundingAccount,
+  FundingAccountListInput,
+  FundingAccountListOutput,
+  UpdateFundingAccount,
+} from './dto';
+
+@Injectable()
+export class FundingAccountService {
+  constructor(
+    @Logger('fundingAccount:service') private readonly logger: ILogger,
+    private readonly db: DatabaseService,
+    private readonly config: ConfigService
+  ) {}
+
+  @OnIndex()
+  async createIndexes() {
+    const constraints = [
+      'CREATE CONSTRAINT ON (n:FundingAccount) ASSERT EXISTS(n.id)',
+      'CREATE CONSTRAINT ON (n:FundingAccount) ASSERT n.id IS UNIQUE',
+      'CREATE CONSTRAINT ON (n:FundingAccount) ASSERT EXISTS(n.active)',
+      'CREATE CONSTRAINT ON (n:FundingAccount) ASSERT EXISTS(n.createdAt)',
+      'CREATE CONSTRAINT ON (n:FundingAccount) ASSERT EXISTS(n.owningOrgId)',
+
+      'CREATE CONSTRAINT ON ()-[r:name]-() ASSERT EXISTS(r.active)',
+      'CREATE CONSTRAINT ON ()-[r:name]-() ASSERT EXISTS(r.createdAt)',
+    ];
+    for (const query of constraints) {
+      await this.db.query().raw(query).run();
+    }
+  }
+
+  // helper method for defining permissions
+  permission = (property: string, baseNode: string) => {
```

Comment:
Why are these still being copied around? Please don't do this. Move it to the query helpers.

Reply:
I'm not seeing where this is being used. Am I blind, lol?